controllers/machine.go

```diff
@@ -345,7 +349,7 @@ func (mrc *machineReconcileContext) ensureHardware() (*tinkv1.Hardware, error) {
 	}
 
 	mrc.tinkerbellMachine.Spec.HardwareName = hardware.Name
-	mrc.tinkerbellMachine.Spec.ProviderID = fmt.Sprintf("tinkerbell://%s", hardware.Spec.ID)
+	mrc.tinkerbellMachine.Spec.ProviderID = fmt.Sprintf("tinkerbell://%s", hardware.UID)
```

Comment:
Use of the UID here would cause issues if types are ever backed up and restored (or if resources are "pivoted" to a different Kubernetes cluster). This wasn't an issue with the Tinkerbell when using the Postgres backend because users because users were specifying the UUID when creating the hardware.

Before recommending an alternative approach, I have a few questions to narrow down what supported permutations we expect to exist.

- Would we expect to be able to reference hardware defined in a different namespace to be used for a given Tinkerbell Machine?
- Would we expect the use of hardware defined on a different k8s cluster to be used?
- Would we expect to be able to reference hardware defined using Tinkerbell with the postgresql backend still?

Reply:
Should we make this change in this PR or should I create an another PR?

Reply:
@panktishah26 its up to you

Reply:
@micahhausler thanks for the clarification, it really helps.

I don't think we need to. block the PR on updating the providerID, especially since it would require additional changes to the controllers to support the new pattern.

I would consider getting things updated to use `tinkerbell//{namespace}/{name}` a release blocker, though to avoid users hitting issues related to backup/restore and pivot.